torchgeo/datasets/biomassters.py

```diff
@@ -0,0 +1,344 @@
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new instance of BioMassters dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train or test split
+            sensors: which sensors to consider for the sample
+            as_time_series: whether or not to return all available
+                time-steps or just a single one for a given target location
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        RuntimeError:
+            AssertionError: if ``split`` or ``sensors`` is invalid
+            ImportError: if pandas is not installed
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        self.root = root
+
+        assert (
+            split in self.valid_splits
+        ), f"Please choose one of the valid splits {self.valid_splits}."
+        self.split = split
+
+        assert set(sensors).issubset(
+            set(self.valid_sensors)
+        ), f"Please choose a subset of valid sensors {self.valid_sensors}."
+        self.sensors = sensors
+        self.as_time_series = as_time_series
+        self.download = download
+        self.checksum = checksum
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        self._verify()
+
+        # open metadata csv files
+        self.df = pd.read_csv(os.path.join(self.root, self.metadata_filename))
+
+        # filter sensors
+        self.df = self.df[self.df["satellite"].isin(self.sensors)]
+
+        # filter split
+        self.df = self.df[self.df["split"] == self.split]
+
+        # generate numerical month from filename since first month is September
+        # and has numerical index of 0
+        self.df["num_month"] = (
+            self.df["filename"]
+            .str.split("_", expand=True)[2]
+            .str.split(".", expand=True)[0]
+            .astype(int)
+        )
+
+        # set dataframe index depending on the task for easier indexing
+        if self.as_time_series:
+            self.df["num_index"] = self.df.groupby(["chip_id"]).ngroup()
+        else:
+            filter_df = (
+                self.df.groupby(["chip_id", "month"])["satellite"].count().reset_index()
+            )
+            filter_df = filter_df[filter_df["satellite"] == len(self.sensors)].drop(
+                "satellite", axis=1
+            )
+            # guarantee that each sample has corresponding number of images available
+            self.df = self.df.merge(filter_df, on=["chip_id", "month"], how="inner")
+
+            self.df["num_index"] = self.df.groupby(["chip_id", "month"]).ngroup()
+
+    def __getitem__(self, index: int) -> dict[str, Any]:
```

Comment:
```suggestion
    def __getitem__(self, index: int) -> dict[str, Tensor]:
```
Pretty sure you only return Tensor